sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsHostMemoryStore.scala

```diff
@@ -44,67 +39,60 @@ class RapidsHostMemoryStore(
   private case object Pooled extends AllocationMode(HOST_MEMORY_BUFFER_PAGEABLE_OFFSET)
   private case object Direct extends AllocationMode(HOST_MEMORY_BUFFER_DIRECT_OFFSET)
 
-  // Returns an allocated host buffer and its allocation mode
-  private def allocateHostBuffer(size: Long): (HostMemoryBuffer, AllocationMode) = {
-    // spill to keep within the targeted size
-    val maybeAmountSpilled = synchronousSpill(math.max(maxSize - size, 0))
-    maybeAmountSpilled.foreach { amountSpilled =>
-      if (amountSpilled != 0) {
-        logInfo(s"Spilled $amountSpilled bytes from the host memory store")
-        TrampolineUtil.incTaskMetricsDiskBytesSpilled(amountSpilled)
-      }
-    }
+  override def getMaxSize: Option[Long] = Some(maxSize)
 
-    var buffer: HostMemoryBuffer = null
-    while (buffer == null) {
-      buffer = PinnedMemoryPool.tryAllocate(size)
-      if (buffer != null) {
-        return (buffer, Pinned)
-      }
+  private def allocateHostBuffer(size: Long): Option[(HostMemoryBuffer, AllocationMode)] = {
+    var buffer: HostMemoryBuffer = PinnedMemoryPool.tryAllocate(size)
+    if (buffer != null) {
+      return Some((buffer, Pinned))
+    }
 
-      if (size > pageableMemoryPoolSize) {
-        if (!haveLoggedMaxExceeded) {
-          logWarning(s"Exceeding host spill max of $pageableMemoryPoolSize bytes to accommodate " +
-              s"a buffer of $size bytes. Consider increasing pageable memory store size.")
-          haveLoggedMaxExceeded = true
-        }
-        return (HostMemoryBuffer.allocate(size, false), Direct)
+    if (size > pageableMemoryPoolSize) {
+      if (!haveLoggedMaxExceeded) {
+        logWarning(s"Exceeding host spill max of $pageableMemoryPoolSize bytes to accommodate " +
+            s"a buffer of $size bytes. Consider increasing pageable memory store size.")
+        haveLoggedMaxExceeded = true
       }
+      return Some((HostMemoryBuffer.allocate(size, false), Direct))
+    }
 
-      val allocation = addressAllocator.allocate(size)
-      if (allocation.isDefined) {
-        buffer = pool.slice(allocation.get, size)
-      } else {
-        val targetSize = math.max(currentSize - size, 0)
-        synchronousSpill(targetSize)
-      }
+    val allocation = addressAllocator.allocate(size)
```

Comment:
try to get all of memory from address space pool and THEN go to `HostMemoryBuffer.allocate`, so this should move to ~ line 50, and we shouldn't need to check that size. We would instead blindly allocate if addressAllocator OOMs